### ISSUE TITLE
New Rule: no-infinite-loop-risk

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#44]
## Research Evidence

From empirical LLM bug studies: "Memory Errors" (infinite loops/recursions) are a documented LLM bug pattern.

## Description

Detect patterns that could lead to infinite loops: while loops without clear exit conditions, recursive calls without base case guards.

### Bad
```ts
// while(true) without break
while (true) {
  processItem();
}

// Recursive without base case
function countdown(n: number) {
  countdown(n - 1);
}

// Loop with condition that can't change
let x = 0;
while (x < 10) {
  console.log(x);
  // Forgot to increment x
}
```

### Good
```ts
while (queue.length > 0) {
  const item = queue.pop();
  process(item);
}

function countdown(n: number) {
  if (n <= 0) return; // Base case
  countdown(n - 1);
}
```

## Detection Heuristics
- `while (true)` without `break` or `return`
- Recursive calls without conditional guard
- Loop variables that aren't modified in loop body

## Notes
- Will have false positives (intentional infinite loops)
- Focus on obvious cases where LLM forgot the exit condition
- Could offer suggestion to add `break` or modify condition

## Priority
Medium - catches dangerous but less common pattern
